src/pages/projects/AddProjectContributorForm.tsx

```diff
@@ -2,35 +2,39 @@ import { Box, Button } from '@mui/material';
 import { useFormikContext } from 'formik';
 import { useState } from 'react';
 import { useTranslation } from 'react-i18next';
+import { useDispatch } from 'react-redux';
 import { ContributorSearchField } from '../../components/ContributorSearchField';
 import { StyledRightAlignedFooter } from '../../components/styled/Wrappers';
-import {
-  CristinProject,
-  ProjectContributor,
-  ProjectContributorType,
-  ProjectFieldName,
-} from '../../types/project.types';
+import { setNotification } from '../../redux/notificationSlice';
+import { CristinProject, ProjectFieldName } from '../../types/project.types';
 import { CristinPerson } from '../../types/user.types';
 import { dataTestId } from '../../utils/dataTestIds';
+import { addContributor, AddContributorErrors } from '../project/helpers/projectContributorHelpers';
 
 interface AddProjectContributorFormProps {
   toggleModal: () => void;
-  addContributor: (
-    personToAdd: CristinPerson | undefined,
-    contributors: ProjectContributor[],
-    roleToAddTo: ProjectContributorType
-  ) => ProjectContributor[] | undefined;
 }
 
-export const AddProjectContributorForm = ({ toggleModal, addContributor }: AddProjectContributorFormProps) => {
+export const AddProjectContributorForm = ({ toggleModal }: AddProjectContributorFormProps) => {
   const { t } = useTranslation();
+  const dispatch = useDispatch();
   const { values, setFieldValue } = useFormikContext<CristinProject>();
   const { contributors } = values;
   const [searchTerm, setSearchTerm] = useState('');
   const [selectedPerson, setSelectedPerson] = useState<CristinPerson>();
 
   const addParticipant = () => {
-    const newContributors = addContributor(selectedPerson, contributors, 'ProjectParticipant');
+    const { newContributors, error } = addContributor(selectedPerson, contributors, 'ProjectParticipant');
+
+    if (error === AddContributorErrors.SAME_ROLE_WITH_SAME_AFFILIATION) {
+      dispatch(
+        setNotification({
+          message: t('project.error.contributor_already_added_with_same_role_and_affiliation'),
+          variant: 'error',
+        })
+      );
+      return;
```

Comment:
Burde vi ikke ha returnert her uansett om det er en error? Nå vil den fortsette om det er en annen error enn `AddContributorErrors.SAME_ROLE_WITH_SAME_AFFILIATION`

Reply:
Ja jeg endrer nok dette når jeg skriver flere tester. I opprinnelig kode returnerte vi aldri error, vi bare returnerte ingenting eller dispatchet en feilmelding. Siden denne funksjonen ikke har tilgang på dispatch nå, må den heller returnere error, så jeg tenker den skal gjøre det hver gang den returnerer tomt nå.

Reply:
Veldig WIP med andre ord, men det er for å dele opp PR'en. Det vises jo ikke noe sted uansett.

Reply:
Skjønner, fint det!